### PR TITLE
fix(perf): enable compile cache

### PR DESCRIPTION
**Pull request by H4ad**
Adding support to [module.enableCompileCache](https://nodejs.org/docs/v22.11.0/api/module.html#moduleenablecompilecachecachedir), added by  https://github.com/nodejs/node/pull/54501, Typescript got [2.5x perf on tsc](https://devblogs.microsoft.com/typescript/announcing-typescript-5-7-beta/#support-for-v8-compile-caching-in-node.js).

On my machine, for the command `--version`, I got the following results:

```bash
Benchmark 1: node bin/npm-cli.js --version
  Time (mean ± σ):      58.0 ms ±   1.1 ms    [User: 51.8 ms, System: 16.8 ms]
  Range (min … max):    56.5 ms …  62.6 ms    51 runs
 
Benchmark 2: node bin/npm-cli-with-cache.js --version
  Time (mean ± σ):      52.9 ms ±   1.0 ms    [User: 46.5 ms, System: 16.8 ms]
  Range (min … max):    51.4 ms …  55.5 ms    56 runs
 
Summary
  'node bin/npm-cli-with-cache.js --version' ran
    1.10 ± 0.03 times faster than 'node bin/npm-cli.js --version'
```

I run the same test with `npm run empty` and I got the same 10% improvement.

### CODE REVIEW

File: lib/cli.js [hunk 1]
```diff
@@ -1,3 +1,10 @@
+try {
+  const { enableCompileCache } = require('node:module')
+  if (enableCompileCache) {
+    enableCompileCache()
+  }
+} catch (e) { /* empty */ }
```
Comment (H4ad) [inline lib/cli.js]: To be safe, I followed the same implementation on `tsc`, ref: https://github.com/microsoft/TypeScript/blob/ef802b1e4ddaf8d6e61d6005614dd796520448f8/Herebyfile.mjs#L332-L337

Comment (wraithgar) [reply lib/cli.js]: We'll need to ignore code coverage on this.

Comment (H4ad) [reply lib/cli.js]: What I should change to make that code ignored?

Comment (wraithgar) [reply lib/cli.js]:
```
/* istanbul ignore next */
```

**Review by wraithgar — APPROVED** [review]

**Comment by wraithgar** [comment]
Thanks!  If you want to duplicate this PR in the `release/v10` branch it could go out w/ the imminent v10 release.

**Comment by H4ad** [comment]
Sure, #7924

**Pull request merged**